weld/subsystem/src/main/java/org/jboss/as/weld/WeldExtension.java

```diff
@@ -119,6 +121,11 @@ protected boolean rejectAttribute(PathAddress address, String attributeName, Mod
                         WeldResourceDefinition.DEVELOPMENT_MODE_ATTRIBUTE)
                  // if the attribute was not discarded it means that it is defined as 'true'. Therefore, reject.
                 .addRejectCheck(RejectAttributeChecker.DEFINED, WeldResourceDefinition.DEVELOPMENT_MODE_ATTRIBUTE)
+                // Discard thread-pool-size attribute - not supported in older versions
+                .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(new ModelNode(WeldExecutorServices.DEFAULT_BOUND)),
+                        WeldResourceDefinition.THREAD_POOL_SIZE_ATTRIBUTE)
```

Comment:
If there is no default value this can be 

.setDiscard(DiscardAttributeChecker.UNDEFINED,
 +                        WeldResourceDefinition.THREAD_POOL_SIZE_ATTRIBUTE)